cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -245,4 +216,14 @@ private boolean matches(String[] actual, String... expected) {
     }
     return true;
   }
+
+  private boolean endsWith(String[] actual, String... expectedEnd) {
+    for (int i = expectedEnd.length - 1; i >= 0; i--) {
```

Comment:
I think you have to guard against the case where actual.length < expectedEnd.length. Otherwise this will attempt to access a negative index and throw.